### PR TITLE
fix: adjusted backButtonInCustomView condition

### PR DESCRIPTION
**Pull request by alduzy**
**Motivation**

This PR intents to fix the back button visibility on Android. On iOS it works as expected, this change would make the behaviour consistent across platforms.

**Screenshots**

I created a snack (you can find it in the "Testplan" section below) that showcases the possible options and found 2 cases failing. All the other cases work fine before and after the change.

```
headerBackVisible: undefined,
headerLeft: () => <Button title="Custom Left" />,
headerTitle: () => <Text>Custom title</Text>
```

<table>
<tr>
<td align="center">
Before ❌ 
</td>
<td align="center">
After ✅ 
</td>
</tr>
<tr>
<td>

Screenshot 2024-09-30 at 13 00 04
  
</td>
<td>

Screenshot 2024-09-30 at 12 59 29

</td>
</tr>
</table>

```
headerBackVisible: true,
headerTitle: () => <Text>Custom title</Text>
```

<table>
<tr>
<td align="center">
Before ❌ 
</td>
<td align="center">
After ✅ 
</td>
</tr>
<tr>
<td>

Screenshot 2024-09-30 at 14 37 12
  
</td>
<td>

Screenshot 2024-09-30 at 14 37 39

</td>
</tr>
</table>

**Test plan**

Use the following code to test this change, set `headerBackVisible` option however you want:
<details>
  <summary>Snack</summary>
```
import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  createNativeStackNavigator,
  NativeStackNavigationProp,
} from '@react-navigation/native-stack';
import { Button, Text } from 'react-native';

type RootStackNavigatorParamsList = {
  Home: undefined;
  Details: undefined;
  DetailsWithLeft: undefined;
  DetailsWithTitle: undefined;
  DetailsWithLeftAndTitle: undefined;
};

const Stack = createNativeStackNavigator<RootStackNavigatorParamsList>();

const HomeScreen = ({
  navigation,
}: {
  navigation: NativeStackNavigationProp<RootStackNavigatorParamsList>;
}) => {
  return (
    <>
      <Button title="Details" onPress={() => navigation.navigate('Details')} />
      <Button
        title="Details with custom left"
        onPress={() => navigation.navigate('DetailsWithLeft')}
      />
      <Button
        title="Details with custom title"
        onPress={() => navigation.navigate('DetailsWithTitle')}
      />
      <Button
        title="Details with custom left and title"
        onPress={() => navigation.navigate('DetailsWithLeftAndTitle')}
      />
    </>
  );
};

const DetailsScreen = ({
  navigation,
}: {
  navigation: NativeStackNavigationProp<RootStackNavigatorParamsList>;
}) => {
  return <Button title="Go to Home" onPress={navigation.goBack} />;
};

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator screenOptions={{ headerBackVisible: false }}>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Details" component={DetailsScreen} />
        <Stack.Screen
          name="DetailsWithLeft"
          component={DetailsScreen}
          options={{
            title: 'Details',
            headerLeft: () => <Button title="Custom Left" />,
          }}
        />
        <Stack.Screen
          name="DetailsWithTitle"
          component={DetailsScreen}
          options={{
            headerTitle: () => <Text>Custom title</Text>,
          }}
        />
        <Stack.Screen
          name="DetailsWithLeftAndTitle"
          component={DetailsScreen}
          options={{
            headerLeft: () => <Button title="Custom Left" />,
            headerTitle: () => <Text>Custom title</Text>,
          }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

```
</details>

### CODE REVIEW

File: packages/native-stack/src/views/HeaderConfig.tsx [hunk 1]
```diff
@@ -165,8 +165,11 @@ export function HeaderConfig({
    * - If `headerTitle` for Android is specified, so we only need to remove the title and keep the back button
    */
   const backButtonInCustomView = headerBackVisible
-    ? headerLeftElement != null
-    : Platform.OS === 'android' && headerTitleElement != null;
+    ? headerLeftElement != null ||
+      (Platform.OS === 'android' && headerTitleElement != null)
+    : Platform.OS === 'android' &&
+      headerTitleElement != null &&
+      headerLeftElement == null;
```
Comment (kkafar) [inline packages/native-stack/src/views/HeaderConfig.tsx:172]: Not gonna lie, this is hard to follow, but I've tested the behaviour & it seems to work nicely.

The `headerBackVisible` is respected (`true` / `false` / `undefined`), presence of `headerLeft` removes back button unless explicitly stated otherwise on both archs. 

Good job :tada:

**Review by kkafar — APPROVED** [review]
Not gonna lie, the condition for `backButtonInCustomView` is hard to follow, but I've tested the behaviour & it seems to work nicely.

The headerBackVisible is respected (true / false / undefined), presence of headerLeft removes back button unless explicitly stated otherwise on both archs.

Good job 🎉

**Comment by alduzy** [comment]
Encouraged by @kkafar I managed to simplify the condition. I've tested the new change in various scenarios using the code snack attached in the description and found no problems.

**Review by kkafar — APPROVED** [review]
LGTM
